server/entity/lightning.go

```diff
@@ -0,0 +1,123 @@
+package entity
+
+import (
+	"github.com/df-mc/dragonfly/server/entity/damage"
+	"github.com/df-mc/dragonfly/server/entity/physics"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"sync/atomic"
+	"time"
+)
+
+// Lightning is a lethal element to thunderstorms. Lightning momentarily increases the skylight's brightness to slightly greater than full daylight.
+type Lightning struct {
+	pos atomic.Value
+
+	state    int
+	liveTime int
+}
+
+// NewLightning creates a lightning entity. The lightning entity will be positioned at the position passed.
+func NewLightning(pos mgl64.Vec3) *Lightning {
+	li := &Lightning{
+		state:    2,
+		liveTime: nextInt(3) + 1,
+	}
+	li.pos.Store(pos)
+
+	return li
+}
+
+// Position returns the current position of the lightning entity.
+func (li *Lightning) Position() mgl64.Vec3 {
+	return li.pos.Load().(mgl64.Vec3)
+}
+
+// World returns the world that the lightning entity is currently in, or nil if it is not added to a world.
+func (li *Lightning) World() *world.World {
+	w, _ := world.OfEntity(li)
+	return w
+}
+
+// AABB ...
+func (Lightning) AABB() physics.AABB {
+	return physics.NewAABB(mgl64.Vec3{}, mgl64.Vec3{})
+}
+
+// State ...
+func (li *Lightning) State() int {
```

Comment:
Is there any use for this to be exported? If not, just remove the method.

Reply:
No, I'll remove it.